core/tee/tee_pobj.c

```diff
@@ -37,20 +37,21 @@ TAILQ_HEAD_INITIALIZER(tee_pobjs);
 static TEE_Result tee_pobj_check_access(uint32_t oflags, uint32_t nflags)
 {
 	/* meta is exclusive */
-	if (oflags | TEE_DATA_FLAG_ACCESS_WRITE_META ||
-	    nflags | TEE_DATA_FLAG_ACCESS_WRITE_META)
-		return TEE_ERROR_ACCESS_CONFLICT;
-
-	if (oflags | TEE_DATA_FLAG_ACCESS_READ &&
-	    !((nflags | TEE_DATA_FLAG_SHARE_READ) &&
-	      oflags | TEE_DATA_FLAG_SHARE_READ))
-		return TEE_ERROR_ACCESS_CONFLICT;
-
-	if (oflags | TEE_DATA_FLAG_ACCESS_WRITE &&
-	    !((nflags | TEE_DATA_FLAG_SHARE_WRITE) &&
-	      oflags | TEE_DATA_FLAG_SHARE_WRITE))
+	if (oflags & TEE_DATA_FLAG_ACCESS_WRITE_META)
 		return TEE_ERROR_ACCESS_CONFLICT;
 
+	/*second open/create need set flags to TEE_DATA_FLAG_SHARE_READ */
+	if (((oflags & TEE_DATA_FLAG_ACCESS_READ) ||
```

Comment:
I would propose the following instead:

	/*
	 * Excerpt of TEE Internal Core API Specification v1.1:
	 * If more than one handle is opened on the same  object, and if any
	 * of these object handles was opened with the flag
	 * TEE_DATA_FLAG_ACCESS_READ, then all the object handles MUST have been
	 * opened with the flag TEE_DATA_FLAG_SHARE_READ
	 */
	if (((oflags & TEE_DATA_FLAG_ACCESS_READ) ||
	     (nflags & TEE_DATA_FLAG_ACCESS_READ)) &&
	    !((nflags & TEE_DATA_FLAG_SHARE_READ) &&
	      (oflags & TEE_DATA_FLAG_SHARE_READ)))
		return TEE_ERROR_ACCESS_CONFLICT;

and the same for the WRITE access.

What do you think?